### PR TITLE
Discount Code Support 

### PR DESCRIPTION
**Pull request by darol100**
This PR adds the ability of adding a discount code in the checkout. See more at: 

https://github.com/Shopify/js-buy-sdk/issues/467

### CODE REVIEW

**Comment by darol100** [comment]
Oops, I have some aggressive linting on my editor that will autoformat everything on save. Let me revert the readme to the old format.

**Comment by darol100** [comment]
@minasmart, I change the README.md to have the original format.

**Comment by darol100** [comment]
Hey @StefanSlehta,

Thanks for reviewing my PR. I have made all changes that you requested.

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and provides the ability to retrieve products and collections from your shop, ad
   + [Updating Line Items](#updating-line-items)
   + [Removing Line Items](#removing-line-items)
   + [Fetching a Checkout](#fetching-a-checkout)
+  + [Adding a discount](#adding-a-discount)
```
Comment (minasmart) [inline README.md]: this is a nitpick, but capitalize `Discount`

File: README.md [hunk 2]
```diff
@@ -181,6 +182,18 @@ client.checkout.fetch(checkoutId).then((checkout) => {
 });
 ```
 
+### Adding a Discount
+```javascript
+const checkoutId = 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTgyMTc3ODc1OTI='; // ID of an existing checkout
+const discountCode = 'best-discount-ever';
+
+// Add an item to the checkout
```
Comment (minasmart) [inline README.md]: this isn't adding an item to the checkout. This is adding a discount to the checkout

**Comment by darol100** [comment]
@minasmart,  I have made those changes that you requested.

**Comment by darol100** [comment]
I rename the graphql file to have the word "Mutation" on it. Just to be consistent with the other mutations.

**Comment by minasmart** [comment]
Thanks! sorry, I haven't had any time to re look at this. I should be able to test it locally and merge in the next few days.

Thanks again!!

**Comment by vlad-dewitt** [comment]
Hello! When will you be able to finish this? Thank you so much!

**Pull request merged**